MAVProxy/modules/mavproxy_paraedit/param_editor_frame.py

```diff
@@ -0,0 +1,241 @@
+        # end wxGlade
+
+    def __set_properties(self):
+        # begin wxGlade: ParamEditor.__set_properties
+        self.SetTitle(("frame_1"))
+        self.SetSize((708, 519))
+        self.display_list.CreateGrid(10, 5)
+        self.display_list.SetColLabelValue(0, ("Param"))
+        self.display_list.SetColLabelValue(1, ("Value"))
+        self.display_list.SetColLabelValue(2, ("Units"))
+        self.display_list.SetColLabelValue(3, ("Options"))
+        self.display_list.SetColLabelValue(4, ("Description"))
+        # end wxGlade
+
+    def __do_layout(self):
+        # begin wxGlade: ParamEditor.__do_layout
+        sizer_1 = wx.BoxSizer(wx.VERTICAL)
+        sizer_2 = wx.BoxSizer(wx.VERTICAL)
+        sizer_3 = wx.BoxSizer(wx.VERTICAL)
+        sizer_4 = wx.BoxSizer(wx.HORIZONTAL)
+        sizer_5 = wx.BoxSizer(wx.HORIZONTAL)
+        sizer_6 = wx.BoxSizer(wx.HORIZONTAL)
+        sizer_6.Add(self.read_file, 0, wx.ALIGN_CENTER, 0)
+        sizer_6.Add(self.write_file, 0, wx.ALIGN_CENTER, 0)
+        sizer_6.Add(self.reset_params, 0, wx.ALIGN_CENTER, 0)
+        sizer_3.Add(sizer_6, 1, wx.ALIGN_CENTER | wx.FIXED_MINSIZE, 0)
+        sizer_5.Add(self.read_params, 0, wx.ALIGN_CENTER, 0)
+        sizer_5.Add(self.write_params, 0, wx.ALIGN_CENTER, 0)
+        sizer_5.Add(self.refresh, 0, wx.ALIGN_CENTER, 0)
+        sizer_3.Add(sizer_5, 1, 0, 0)
+        sizer_4.Add(self.search_key, 0, wx.ALIGN_CENTER, 0)
+        sizer_4.Add(self.search, 0, wx.ALIGN_CENTER_VERTICAL | wx.ALIGN_RIGHT, 0)
+        sizer_3.Add(sizer_4, 1, 0, 0)
+        sizer_2.Add(sizer_3, 1, wx.ALIGN_CENTER | wx.FIXED_MINSIZE, 0)
+        sizer_2.Add(self.display_list, 1, wx.EXPAND, 0)
+        sizer_1.Add(sizer_2, 1, wx.ALIGN_CENTER | wx.FIXED_MINSIZE, 0)
+        self.SetSizer(sizer_1)
+        self.Layout()
+        # end wxGlade
+
+    def set_event_queue(self, q):
+        self.event_queue = q
+
+    def set_event_queue_lock(self, l):
+        self.event_queue_lock = l
+
+    def set_gui_event_queue(self, q):
+        self.gui_event_queue = q
+
+    def set_gui_event_queue_lock(self, l):
+        self.gui_event_queue_lock = l
+
+    def set_close_window_semaphore(self, sem):
+        self.close_window_semaphore = sem
+
+    def time_to_process_gui_events(self, evt):
+        event_processed = False
+        queue_access_start_time = time.time()
+        self.gui_event_queue_lock.acquire()
+        while self.gui_event_queue.qsize() > 0 and (time.time() < queue_access_start_time) < 0.6:
+            event_processed = True
+            event = self.gui_event_queue.get()
+            try:
+                self.process_gui_event(event)
+            except Exception as e:
+                print("Caught exception (%s)" % str(e))
+
+        self.gui_event_queue_lock.release()
+
+        if (event_processed == True):
+            #redraw window to apply changes
+            self.Refresh()
+            self.Update()
+
+    def process_gui_event(self, event):
```

Comment:
@tridge @peterbarker Can you please look into why this function is not being executed? I have used most of the stuff from mission editor module to make GUI.

Reply:
Figured it out. Missed the timer in initialisation. Sorry for the inconvenience.